grype/db/v5/namespace/index_test.go

```diff
@@ -137,124 +138,155 @@ func TestIndex_NamespacesForDistro(t *testing.T) {
 	assert.NoError(t, err)
 
 	tests := []struct {
+		name       string
 		distro     *osDistro.Distro
 		namespaces []*distro.Namespace
 	}{
 		{
+			name:   "alpine patch version matches minor version namespace",
 			distro: newDistro(t, osDistro.Alpine, "3.15.4", []string{"alpine"}),
 			namespaces: []*distro.Namespace{
 				distro.NewNamespace("alpine", osDistro.Alpine, "3.15"),
 			},
 		},
 		{
+			name:   "alpine minor version matches minor version namespace",
 			distro: newDistro(t, osDistro.Alpine, "3.16", []string{}),
 			namespaces: []*distro.Namespace{
 				distro.NewNamespace("alpine", osDistro.Alpine, "3.16"),
 			},
 		},
 		{
+			name:   "alpine edge version matches edge namespace",
+			distro: &osDistro.Distro{Type: osDistro.Alpine, Version: nil, RawVersion: "3.17_alpha20221002", IDLike: []string{"alpine"}},
```

Comment:
we should also add a case that covers the possibility if they start changing their versions to be semver compliant (or close) as an extra precaution (e.g. `3.17-alpha20221002` or `3.17.0-alpha20221002`). I realize this is unlikely, but knowing the namespace logic is resilient in cases like these is useful since the data can change out from underneath us (and the alternative is a code change when that happens).

Reply:
Test cases were added for this possibility to check for namespace resilience.